File: src/main/java/com/methodpark/cmpe/code/isp/GermanFileTranslator.java
package com.methodpark.cmpe.code.isp;

import java.io.BufferedReader;
import java.io.FileReader;
import java.util.HashMap;

public class GermanFileTranslator implements IFileTranslator {
    
    HashMap<String,String> dict = new HashMap<String, String>();
    
    public GermanFileTranslator()
    {
        dict.put("cat", "Katze");
        dict.put("dog", "Hund");
        dict.put("house", "Haus");
        dict.put("duck", "Ente");
    }

    public String translate(String text) {
      String ret = dict.get(text);
      if (ret != null)
          return ret;
      else
          return text;
    }

    public String loadFile(String filename) {
        BufferedReader reader = null;
        try
        {
            reader = new BufferedReader(new FileReader(filename));

            String ret = "";
            String line = null;
            while ((line = reader.readLine()) != null) {
                ret += line;
            }
            reader.close();
            return ret;
        }
        catch (Exception e)
        {
            e.printStackTrace();
        }
        return "";
    }
}

File: src/main/csharp/Isp/Isp/GermanFileTranslator.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace Isp
{
    public class GermanFileTranslator : IFileTranslator
    {
        private readonly Dictionary<String, String> dict = new Dictionary<String, String>();

        public GermanFileTranslator()
        {
            dict.Add("cat", "Katze");
            dict.Add("dog", "Hund");
            dict.Add("house", "Haus");
            dict.Add("duck", "Ente");
        }

        public String Translate(String text)
        {
            return dict.ContainsKey(text) ? dict[text] : text;
        }

        public String LoadFile(String filename)
        {
            return File.ReadAllText(filename);
        }
    }
}